types/tx.go

```diff
@@ -161,15 +161,16 @@ func ComputeProtoSizeForTxs(txs []Tx) int64 {
 	return int64(pdData.Size())
 }
 
-// DecodeChildTx attempts to unmarshal the provided transaction into a child
+// UnwrapChildTx attempts to unmarshal the provided transaction into a child
 // transaction wrapper, if this an be done, then it returns true. A child
-// transaction is a normal transaction that has been derived from a different
-// parent transaction. The returned hash is that of the parent transaction,
-// which allows us to remove the parent transaction from the mempool. NOTE:
-// protobuf sometimes does not throw an error if the transaction passed is not
-// a tmproto.ChildTx, since the schema for PayForMessage is kept in the app, we
-// cannot perform further checks without creating an import cycle.
-func DecodeChildTx(tx Tx) (hash []byte, unwrapped Tx, has bool) {
+// transaction is a normal transaction that has been derived (malleated) from a
+// different parent transaction. The returned hash is that of the parent
+// transaction, which allows us to remove the parent transaction from the
+// mempool. NOTE: protobuf sometimes does not throw an error if the transaction
+// passed is not a tmproto.ChildTx, since the schema for PayForMessage is kept
+// in the app, we cannot perform further checks without creating an import
+// cycle.
+func UnwrapChildTx(tx Tx) (parentHash []byte, unwrapped Tx, isChild bool) {
```

Comment:
Maybe orthogonal to this PR, but semantically should this not return an error instead of a Boolean?